README.markdown

```diff
@@ -436,16 +436,27 @@ scripts.
 
 __4.12. Q. How can I check scripts written for different versions of Ruby?__
 
-A. Install a Ruby version manager such as [rvm][rvm] or [rbenv][rbenv],
-activate the environment for the relevant version of Ruby, and install in it
-the checkers you want to use.  Set `g:syntastic_ruby_checkers` accordingly in
-your `vimrc`, and run [Vim][vim] from the virtual environment.
-
-If you're starting Vim from a desktop manager rather than from a terminal you
-might need to write wrapper scripts around your checkers, to activate the
-virtual environment before running the actual checks.  Then you'll need to
-point the relevant `g:syntastic_ruby_<checker>_exec` variables to the wrapper
-scripts.
+A. Install a Ruby version manager such as [rvm][rvm],  [rbenv][rbenv],
+[chruby][chruby] or [ry][ry], activate the relevant version of Ruby, and
```

Comment:
> Making environments that survive all the way to programs run from Vim is less than trivial, to say the least. Did you actually check that chruby and ry solve the problem in question on at least Windows, Mac OS X, and Linux?

You are right, this is a tricky problem, hence the plurality of tools. They do solve the problem: I personally use `chruby` and a coworker uses `ry` daily on the relevant platforms. Both are in wide enough use to be considered appropriate for the task, each one catering to different personal workflows or opinions:

- rvm overloads the shell's `cd` builtin
- rbenv generates and uses shimming shell wrappers for each gem executable script, doing away with the cd thing and leaving your env untouched, but the shims can get out of sync
- chruby leverages environment variables (mostly `PATH`, `GEM_*` and `RUBY_*`)
- ry leverages symlinks and doesn't touch anything else (env or otherwise), but thus acts globally

URLs for the record:

- https://github.com/postmodern/chruby
- https://github.com/jneen/ry